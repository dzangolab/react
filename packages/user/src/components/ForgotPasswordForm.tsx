import {
  Email,
  Provider,
  FormActions,
  emailSchema,
} from "@dzangolab/react-form";
import { useTranslation } from "@dzangolab/react-i18n";
import { AuthFormWrapper } from "@dzangolab/react-ui";
import React from "react";
import * as zod from "zod";

interface Properties {
  handleSubmit: (email: string) => void;
  links?: React.ReactNode;
  loading?: boolean;
}

export const ForgotPasswordForm = ({
  handleSubmit,
  links,
  loading,
}: Properties) => {
  const { t } = useTranslation("user");

  const ForgotPasswordFormSchema = zod.object({
    email: emailSchema({
      required: t("validation.messages.email"),
      invalid: t("validation.messages.validEmail"),
    }),
  });

  const renderForm = () => {
    return (
      <Provider
        validationSchema={ForgotPasswordFormSchema}
        onSubmit={(data) => handleSubmit(data.email)}
      >
        <Email
          label={t("forgotPassword.form.email.label")}
          name="email"
          placeholder={t("forgotPassword.form.email.placeholder")}
        />

        <FormActions
          actions={[
            {
              id: "submit",
              label: t("forgotPassword.form.actions.submit"),
            },
          ]}
          loading={loading}
          alignment="fill"
        />
      </Provider>
    );
  };

  return (
    <AuthFormWrapper
      className="forgot-password"
      form={renderForm()}
      links={links}
    />
  );
};
