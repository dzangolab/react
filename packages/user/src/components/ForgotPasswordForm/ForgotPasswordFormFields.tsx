import { FormActions, Email, useFormContext } from "@prefabs.tech/react-form";
import { useTranslation } from "@prefabs.tech/react-i18n";
import React from "react";

interface Properties {
  loading?: boolean;
}

export const ForgotPasswordFormFields = ({ loading }: Properties) => {
  const { t } = useTranslation("user");

  const {
    // eslint-disable-next-line @typescript-eslint/no-unused-vars
    formState: { errors, submitCount, isDirty },
  } = useFormContext();

  return (
    <>
      <Email
        label={t("forgotPassword.form.email.label")}
        name="email"
        placeholder={t("forgotPassword.form.email.placeholder")}
      />

      <FormActions
        actions={[
          {
            id: "submit",
            label: t("forgotPassword.form.actions.submit"),
            disabled: !isDirty,
          },
        ]}
        loading={loading}
        alignment="fill"
      />
    </>
  );
};
