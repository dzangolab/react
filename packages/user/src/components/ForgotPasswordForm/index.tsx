import { Provider, emailSchema } from "@prefabs.tech/react-form";
import { useTranslation } from "@prefabs.tech/react-i18n";
import React from "react";
import * as zod from "zod";

import { ForgotPasswordFormFields } from "./ForgotPasswordFormFields";

interface Properties {
  handleSubmit: (email: string) => void;
  loading?: boolean;
}

export const ForgotPasswordForm = ({ handleSubmit, loading }: Properties) => {
  const { t, i18n } = useTranslation("user");

  const ForgotPasswordFormSchema = zod.object({
    email: emailSchema({
      required: t("validation.messages.email"),
      invalid: t("validation.messages.validEmail"),
    }),
  });

  return (
    <Provider
      validationSchema={ForgotPasswordFormSchema}
      onSubmit={(data) => handleSubmit(data.email)}
      validationTriggerKey={i18n.language}
    >
      <ForgotPasswordFormFields loading={loading} />
    </Provider>
  );
};
