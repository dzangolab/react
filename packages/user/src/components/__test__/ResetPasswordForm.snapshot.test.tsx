import { render } from "@testing-library/react";
import React from "react";
import { expect, test } from "vitest";

import { ForgotPasswordForm } from "../ForgotPasswordForm";

test("Component matches snapshot", () => {
  const { container } = render(
    // eslint-disable-next-line @typescript-eslint/no-empty-function
    <ForgotPasswordForm handleSubmit={() => {}} />,
  );

  expect(container).toMatchSnapshot();
});
