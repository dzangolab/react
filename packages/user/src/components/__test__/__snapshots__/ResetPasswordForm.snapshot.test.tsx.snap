// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Component matches snapshot 1`] = `
<form
  className=""
  noValidate={true}
  onSubmit={[Function]}
>
  <div
    className="field email"
  >
    <label
      htmlFor="input-field-email"
    >
      forgetPassword.form.email.label
    </label>
    <input
      id="input-field-email"
      name="email"
      onBlur={[Function]}
      onChange={[Function]}
      placeholder="forgetPassword.form.email.placeholder"
      readOnly={false}
      type="email"
    />
  </div>
  <button
    className="dz-button button-submit primary filled medium"
    type="submit"
  >
    <span
      className="label"
    >
      forgetPassword.form.actions.submit
    </span>
  </button>
</form>
`;
